tests/python/pants_test/engine/test_test.py

```diff
@@ -0,0 +1,72 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from future.utils import text_type
+
+from pants.build_graph.address import Address
+from pants.engine.console import Console
+from pants.engine.legacy.graph import HydratedTarget
+from pants.engine.rules import console_rule
+from pants.engine.selectors import Get, Select
+from pants.util.meta import AbstractClass
+from pants.util.objects import datatype
+from pants_test.engine.scheduler_test_base import SchedulerTestBase
+from pants_test.engine.util import MockConsole, run_rule
+from pants_test.test_base import TestBase
+
+
+# TODO: Move to production code
+class Status(object):
+  SUCCESS = "SUCCESS"
+  FAILURE = "FAILURE"
+
+
+# TODO: Pretty sure these shouldn't be text_types.
+# status should be an enum somehow, and stdout should be...
+# I guess text_type, actually... But also maybe bytes?
+# When do we start asserting that testrunner output is unicode?
+class TestResult(datatype([
+  # One of the Status pseudo-enum values capturing whether the run was successful.
+  ('status', text_type),
+  # The stdout of the test runner (which may or may not include actual testcase output).
+  ('stdout', text_type)
+])):
+  pass
+
+
+# TODO: Move to production code
+@console_rule('test', [Select(Console), Select(TestResult)])
+def fast_test(console, hydrated_targets):
+  test_results = yield [Get(TestResult, HydratedTarget, ht) for ht in hydrated_targets]
+  wrote_any_stdout = False
+  for test_result in test_results:
+    wrote_any_stdout |= bool(test_result.stdout)
+    # Assume \n-terminated
+    console.write_stdout(test_result.stdout)
```

Comment:
I have so far been assuming (and I think it was in the v2 HLD) that we would render test runs completing as soon as they did. So the test stderr/stdout rendering would happen in #6003 by writing to a logger, and the test would "slow fail"... return a failing test result.

Reply:
I _thought_ the HLD said we would aggregate, and that we would run the thing running the test itself in some kind of interactive mode if we wanted things more eagerly...

I also think we started talking about introducing a "select first" version of `Get` so that we could be more eager if we wanted to be...